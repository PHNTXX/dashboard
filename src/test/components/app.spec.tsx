import { render } from "@testing-library/react";
import App, { IAppProps } from "../../components/app";

const props: IAppProps = {
  name: "App Test",
  icon: "bug_report",
  url: "#",
  displayURL: "test",
};

describe("app.tsx", () => {
  it("Tests app rendering with newTab=true", () => {
    const { asFragment } = render(
      <App
        name={props.name}
        icon={props.icon}
        url={props.url}
        displayURL={props.displayURL}
        newTab={true}
      />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("Tests app rendering with newTab=false", () => {
    const { asFragment } = render(
      <App
        name={props.name}
        icon={props.icon}
        url={props.url}
        displayURL={props.displayURL}
        newTab={false}
      />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("Tests app rendering without newTab", () => {
    const { asFragment } = render(
      <App
        name={props.name}
        icon={props.icon}
        url={props.url}
        displayURL={props.displayURL}
      />,
    );

    expect(asFragment).toMatchSnapshot();
  });
});
