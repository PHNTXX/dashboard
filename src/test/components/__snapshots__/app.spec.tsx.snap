// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.tsx Tests app rendering with newTab=false 1`] = `[Function]`;

exports[`app.tsx Tests app rendering with newTab=true 1`] = `[Function]`;

exports[`app.tsx Tests app rendering without newTab 1`] = `[Function]`;
